Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -27,8 +28,8 @@ public record DafnyDocument(
     bool CanDoVerification,
     // VerificationDiagnostics can be deduced from CounterExamples,
     // but they are stored separately because they are migrated and counterexamples currently are not.
-    IReadOnlyDictionary<ImplementationId, ImplementationView>? ImplementationViews,
-    IReadOnlyList<Counterexample> CounterExamples,
+    IReadOnlyDictionary<ImplementationId, ImplementationView> ImplementationViewsView,
```

Comment:
I suggest another naming that is less confusing:
```suggestion
    IReadOnlyDictionary<ImplementationId, ImplementationView> ImplementationIdToView,
```
```suggestion
    IReadOnlyDictionary<ImplementationId, ImplementationView> ImplementationViewsView,
```